### ISSUE TITLE
Volumes mounted incorrectly if the directory is not in the mounts section of the Colima config

### ISSUE THREAD

**Issue by nickcodefresh** [abiosoft/colima#844]
### Description

If a container is started with a volume mount that is not included in the mounts section of `colima.yaml`, then any files in that directory is identified as a directory.

### Version

Colima Version: 0.5.6
Lima Version: 0.18.0
Qemu Version: 8.1.2

### Operating System

- [ ] macOS Intel <= 12 (Monterrey)
- [ ] macOS Intel >= 13 (Ventura)
- [ ] macOS M1 <= 12 (Monterrey)
- [X] macOS M1 >= 13 (Ventura)
- [ ] Linux

### Output of `colima status`

INFO[0000] colima is running using macOS Virtualization.Framework
INFO[0000] arch: aarch64
INFO[0000] runtime: docker
INFO[0000] mountType: virtiofs
INFO[0000] socket: unix:///Users/nicksmith/.colima/default/docker.s

### Reproduction Steps

```
➜   docker run -ti --rm  -v ~/Documents/mydir:/x ubuntu /bin/sh
# cd /x
# ls
total 8
drwxr-xr-x 5 root root  120 Nov  7 09:03 .
drwxr-xr-x 1 root root 4096 Nov  7 09:05 ..
drwxr-xr-x 2 root root   40 Nov  7 08:59 miocm-config.yml
```
Note that the `miocm-config.yml` file has been mounted as a directory.

If I now add that directory into the mount section of my `colima.yaml` file, the files are files.
```
mounts:
  - location: /Users/me/Documents/mydir
    writable: true
```
```
➜   docker run -ti --rm  -v ~/Documents/mydir:/x ubuntu /bin/sh
# cd /x
# ls
total 8
drwxr-xr-x 5 root root  120 Nov  7 09:03 .
drwxr-xr-x 1 root root 4096 Nov  7 09:05 ..
-rwxrwxrwx 1 root root  135 Nov  6 17:17 miocm-config.yml
```

### Expected behaviour

An error should be throw and the container fail.

**Comment by yonghaoy** [comment]
@nickcodefresh Do you have a work around? I had the same issue, and adding the dir in `colima.yaml` does not solve my problem. 
My issue: I have a file  `/tmp/random-blabla-name.json`. And I want to use that it into container when running docker
`docker-compose run -v /tmp/random-blabla-name.json/:/creds/input.json...`
It works with docker desktop but not working with Colima.
I suspect my issue is similar, but adding `/creds/` in `colima.yaml` does not work. 

Also, seems : https://github.com/abiosoft/colima/issues/838 is also related.

**Comment by abiosoft** [comment]
@yonghaoy, you can put it in `/tmp/colima/random-blabla-name.json` instead.

`/tmp/colima` is mounted by default.

**Comment by debo** [comment]
@abiosoft apologies for the tag/summon

I am still having issues with this, the command I am running is the following:
```shell
~ docker run -d -p 8080:8080 -p 80:80 -v /Users/username/.traefik.yaml:/etc/traefik/traefik.yml traefik:v3.3
```
But the logs show the following error:
```shell
{"level":"error","error":"command traefik error: read /etc/traefik/traefik.yml: is a directory","time":"2025-03-06T09:57:45Z","message":"Command error"}
```
The file for some reasons is mounted as directory despite being an existing valid file in my home folder (which I understand should work fine from reading your comments in [this issue](https://github.com/abiosoft/colima/issues/838)). Any idea why this might be happening? I have a workaround at the moment but it's not ideal because it involves having a `docker-compose.yml` and a standalone project from where to launch this.

**Comment by abiosoft** [comment]
@debo can you destroy the container and re-create it? `docker rm -f <container name>` should work fine.

Is the `traefik.yml` something you can share (may be with some details redacted)? I can try to reproduce the error on my end.

**Comment by debo** [comment]
I already did that and also restarted Colima but I am happy to give it another try. 

The yaml file is the vanilla, sample one that you can find [here](https://doc.traefik.io/traefik/getting-started/install-traefik/).

For added information I am running on Apple Silicon and I am running Sequoia 15.3.1.

Worth mentioning that I have an added volume mounted as part of the Colima configuration and it looks ok, however my home folder doesn't look fully mounted. Only the `Library` folder is visible if I ssh into Colima.

Last but not least I tried to put the `traefik.yaml` both in the home and in the additional mount point but the behaviour is the same and I also tried to name the file with and without the leading dot.

**Comment by abiosoft** [comment]
@debo once you specify a custom mount, it overrides the default and your home folder is no longer mounted.
So you would have to include your home folder in the mounts as well.

I think once you do that and recreate the container, it should work as desired.

**Comment by debo** [comment]
Fair enough, with that said though, the fact my home folder is not properly mounted is somehow irrelevant to the issue I would say if I mount it explicitly right?

Anyway I think I found the culprit but I am not sure if it's intended behaviour or not. If I change my command to be:
```shell
~ docker run -d -p 8080:8080 -p 80:80 -v /Users/username/.traefik.yaml:/etc/traefik/ traefik:v3.3
```
Everything seems to work fine. Is that correct and intended behaviour?
